CRM/Utils/Request.php

```diff
@@ -126,8 +126,10 @@ public static function retrieve($name, $type, &$store = NULL, $abort = FALSE, $d
     }
 
     // minor hack for action
-    if ($name == 'action' && is_string($value)) {
-      $value = CRM_Core_Action::resolve($value);
+    if ($name == 'action') {
+      if (!is_numeric($value) && is_string($value)) {
+        $value = CRM_Core_Action::resolve($value);
+      }
     }
```

Comment:
This was not allowing an action to be passed in via a numeric ID, only by name (eg. view/edit etc).  As the action is assigned to the template numerically and we are creating a URL via the template it would be a real pain to map back to string only to map back to ID again.